### ISSUE TITLE
jdk_nio: Agent communication error during java.nio.Buffer test (SIGSEGV crash)

### ISSUE THREAD

**Issue by smlambert** [AdoptOpenJDK/openjdk-tests#57]
Agent crashes during java/nio/Buffer test,  SIGSEGV (0xb) at pc=0x00007f12a483e058, pid=28852, tid=0x00007f128c392700

Problematic frame:
20:21:58 [2017-06-14 18:22:25,715] Agent[0]: stdout: # C  [libjava.so+0xe058]  Java_java_nio_Bits_copyToShortArray+0x1d8

output from https://ci.adoptopenjdk.net/view/OpenJDK%20tests/job/openjdk_test_x86-64_linux/34/consoleFull:

20:21:58 [2017-06-14 18:22:25,714] Agent[0]: stdout: #
20:21:58 [2017-06-14 18:22:25,714] Agent[0]: stdout: # A fatal error has been detected by the Java Runtime Environment:
20:21:58 [2017-06-14 18:22:25,715] Agent[0]: stdout: #
20:21:58 [2017-06-14 18:22:25,715] Agent[0]: stdout: #  SIGSEGV (0xb) at pc=0x00007f12a483e058, pid=28852, tid=0x00007f128c392700
20:21:58 [2017-06-14 18:22:25,715] Agent[0]: stdout: #
20:21:58 [2017-06-14 18:22:25,715] Agent[0]: stdout: # JRE version: OpenJDK Runtime Environment (8.0) (build 1.8.0-internal-jenkins_2017_05_17_18_01-b00)
20:21:58 [2017-06-14 18:22:25,715] Agent[0]: stdout: # Java VM: OpenJDK 64-Bit Server VM (25.71-b00 mixed mode linux-amd64 compressed oops)
20:21:58 [2017-06-14 18:22:25,715] Agent[0]: stdout: # Problematic frame:
20:21:58 [2017-06-14 18:22:25,715] Agent[0]: stdout: # C  [libjava.so+0xe058]  Java_java_nio_Bits_copyToShortArray+0x1d8
20:21:58 [2017-06-14 18:22:25,716] Agent[0]: stdout: #
20:21:58 [2017-06-14 18:22:25,716] Agent[0]: stdout: # Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
20:21:58 [2017-06-14 18:22:25,716] Agent[0]: stdout: #
20:21:58 [2017-06-14 18:22:25,716] Agent[0]: stdout: # An error report file with more information is saved as:
20:21:58 [2017-06-14 18:22:25,716] Agent[0]: stdout: # /home/jenkins/workspace/openjdk_test_x86-64_linux/jvmtest/OpenJDK_Playlist/SE80/work/scratch/0/hs_err_pid28852.log
20:21:58 [2017-06-14 18:22:25,796] Agent[0]: stdout: #
20:21:58 [2017-06-14 18:22:25,796] Agent[0]: stdout: # If you would like to submit a bug report, please visit:
20:21:58 [2017-06-14 18:22:25,796] Agent[0]: stdout: #   http://bugreport.java.com/bugreport/crash.jsp
20:21:58 [2017-06-14 18:22:25,796] Agent[0]: stdout: #
20:21:58 TEST: java/nio/Buffer/Basic.java
20:21:58 TEST JDK: /home/jenkins/workspace/openjdk_test_x86-64_linux/openjdkbinary/j2sdk-image/jre/bin/../..
20:21:58 
20:21:58 ACTION: build -- Passed. All files up to date
20:21:58 REASON: Named class compiled on demand
20:21:58 TIME:   0.002 seconds
20:21:58 messages:
20:21:58 command: build Basic
20:21:58 reason: Named class compiled on demand
20:21:58 elapsed time (seconds): 0.002
20:21:58 
20:21:58 ACTION: main -- Error. Agent communication error: java.io.EOFException; check console log for any additional details
20:21:58 REASON: Assumed action based on file name: run main Basic 
20:21:58 TIME:   4.982 seconds
20:21:58 messages:
20:21:58 command: main Basic
20:21:58 reason: Assumed action based on file name: run main Basic 
20:21:58 Mode: agentvm
20:21:58 Agent id: 0
20:21:58 elapsed time (seconds): 4.982
20:21:58 configuration:
20:21:58 Boot Layer
20:21:58   class path: /home/jenkins/workspace/openjdk_test_x86-64_linux/openjdkbinary/j2sdk-image/jre/bin/../../lib/tools.jar
20:21:58               /home/jenkins/workspace/openjdk_test_x86-64_linux/openjdk-test/TestConfig/lib/jtreg/lib/javatest.jar
20:21:58               /home/jenkins/workspace/openjdk_test_x86-64_linux/openjdk-test/TestConfig/lib/jtreg/lib/jtreg.jar
20:21:58 
20:21:58 Test Layer
20:21:58   class path: /home/jenkins/workspace/openjdk_test_x86-64_linux/jvmtest/OpenJDK_Playlist/SE80/work/classes/1/java/nio/Buffer
20:21:58               /home/jenkins/workspace/openjdk_test_x86-64_linux/openjdk-test/OpenJDK_Playlist/openjdk-jdk8u/jdk/test/java/nio/Buffer
20:21:58 
20:21:58 rerun:
20:21:58 HOME=/home/jenkins \
20:21:58 LANG=en_US.UTF-8 \
20:21:58 PATH=/bin:/usr/bin \
20:21:58     /home/jenkins/workspace/openjdk_test_x86-64_linux/openjdkbinary/j2sdk-image/jre/bin/../../bin/java \
20:21:58         -Dtest.class.path.prefix=/home/jenkins/workspace/openjdk_test_x86-64_linux/jvmtest/OpenJDK_Playlist/SE80/work/classes/1/java/nio/Buffer:/home/jenkins/workspace/openjdk_test_x86-64_linux/openjdk-test/OpenJDK_Playlist/openjdk-jdk8u/jdk/test/java/nio/Buffer \
20:21:58         -Dtest.src=/home/jenkins/workspace/openjdk_test_x86-64_linux/openjdk-test/OpenJDK_Playlist/openjdk-jdk8u/jdk/test/java/nio/Buffer \
20:21:58         -Dtest.src.path=/home/jenkins/workspace/openjdk_test_x86-64_linux/openjdk-test/OpenJDK_Playlist/openjdk-jdk8u/jdk/test/java/nio/Buffer \
20:21:58         -Dtest.classes=/home/jenkins/workspace/openjdk_test_x86-64_linux/jvmtest/OpenJDK_Playlist/SE80/work/classes/1/java/nio/Buffer \
20:21:58         -Dtest.class.path=/home/jenkins/workspace/openjdk_test_x86-64_linux/jvmtest/OpenJDK_Playlist/SE80/work/classes/1/java/nio/Buffer \
20:21:58         -Dtest.vm.opts='-ea -esa -Xmx512m' \
20:21:58         -Dtest.tool.vm.opts='-J-ea -J-esa -J-Xmx512m' \
20:21:58         -Dtest.compiler.opts= \
20:21:58         -Dtest.java.opts= \
20:21:58         -Dtest.jdk=/home/jenkins/workspace/openjdk_test_x86-64_linux/openjdkbinary/j2sdk-image/jre/bin/../.. \
20:21:58         -Dcompile.jdk=/home/jenkins/workspace/openjdk_test_x86-64_linux/openjdkbinary/j2sdk-image/jre/bin/../.. \
20:21:58         -Dtest.timeout.factor=4.0 \
20:21:58         -classpath /home/jenkins/workspace/openjdk_test_x86-64_linux/jvmtest/OpenJDK_Playlist/SE80/work/classes/1/java/nio/Buffer:/home/jenkins/workspace/openjdk_test_x86-64_linux/openjdk-test/OpenJDK_Playlist/openjdk-jdk8u/jdk/test/java/nio/Buffer:/home/jenkins/workspace/openjdk_test_x86-64_linux/openjdkbinary/j2sdk-image/jre/bin/../../lib/tools.jar:/home/jenkins/workspace/openjdk_test_x86-64_linux/openjdk-test/TestConfig/lib/jtreg/lib/javatest.jar:/home/jenkins/workspace/openjdk_test_x86-64_linux/openjdk-test/TestConfig/lib/jtreg/lib/jtreg.jar \
20:21:58         Basic
20:21:58 
20:21:58 TEST RESULT: Error. Agent communication error: java.io.EOFException; check console log for any additional details

**Comment by sophia-guo** [comment]
Duplicate of #118
